lib/util/state.js

```diff
@@ -106,75 +106,116 @@ export function getActiveItineraries (otpState) {
  * FIXME: Do major testing to get this right.
  */
-function calculateItineraryCost (itinerary) {
-  // FIXME: Add drive reluctance factor?
-  // FIXME: Make these configurable.
-  const FARE_COST_FACTOR = 0.5
-  const DURATION_COST_FACTOR = 0.25
-  const WALK_COST_FACTOR = 0.1
-  const WAIT_COST_FACTOR = 0.1
-  const TRANSFER_COST_FACTOR = 0.9
-  return getTotalFare(itinerary) * FARE_COST_FACTOR +
-    itinerary.duration * DURATION_COST_FACTOR +
-    itinerary.walkDistance * WALK_COST_FACTOR +
-    itinerary.waitingTime * WAIT_COST_FACTOR +
-    itinerary.transfers * TRANSFER_COST_FACTOR
+function calculateItineraryCost (itinerary, config = {}) {
+  const weights = {
+    walkReluctance: WALK_RELUCTANCE,
+    driveReluctance: DRIVE_RELUCTANCE,
+    durationFactor: DURATION_FACTOR,
+    fareFactor: FARE_FACTOR,
+    waitReluctance: WAIT_RELUCTANCE,
+    transferReluctance: TRANSFER_RELUCTANCE
+  }
+  // If config contains values to override defaults, apply those.
+  const configWeights = config.itinerary && config.itinerary.weights
+  if (configWeights) Object.assign(weights, configWeights)
+  return getTotalFare(itinerary, config) * weights.fareFactor +
+    itinerary.duration * weights.durationFactor +
+    itinerary.walkDistance * weights.walkReluctance +
+    getDriveTime(itinerary) * weights.driveReluctance +
+    itinerary.waitingTime * weights.waitReluctance +
+    itinerary.transfers * weights.transferReluctance
+}
+
+/**
+ * Get total drive time (i.e., total duration for legs with mode=CAR) for an
+ * itinerary.
+ */
+function getDriveTime (itinerary) {
+  if (!itinerary) return 0
+  let driveTime = 0
+  itinerary.legs.forEach(leg => {
+    if (leg.mode === 'CAR') driveTime += leg.duration
+  })
+  return driveTime
 }
 
+// $2 per trip? This is a made up number.
+const DEFAULT_BIKESHARE_FARE = 2 * 100
+// $2 for 3 hours of parking?
+const DEFAULT_PARKING_COST = 3 * 2.00 * 100
+// FL per diem rate: https://www.flcourts.org/content/download/219314/1981830/TravelInformation.pdf
+const DEFAULT_DRIVING_COST_PER_MILE = 0.445 * 100
+
 // FIXME: Move to otp-ui?
 // TODO: Add GBFS fares
 /**
  * Returns total fare for itinerary (in cents)
  */
-function getTotalFare (itinerary) {
+function getTotalFare (itinerary, config = {}) {
+  // Get transit/TNC fares.
   const {maxTNCFare, transitFare} = calculateFares(itinerary)
-  // TODO: move driving costs to calc fares?
-  // TODO: Add config settings?
-  // $2 for 3 hours of parking?
-  const PARKING_COST = 3 * 2.00 * 100
-  // FL per diem rate: https://www.flcourts.org/content/download/219314/1981830/TravelInformation.pdf
-  const PER_MILE_COST_TO_DRIVE = 0.445 * 100
-  const BIKESHARE_FARE = 2 * 100 // $2 per trip? This is a made up number.
+  // Start with default weighting values.
+  const costs = {
+    bikeshareTripCostCents: DEFAULT_BIKESHARE_FARE,
+    carParkingCostCents: DEFAULT_PARKING_COST,
+    drivingCentsPerMile: DEFAULT_DRIVING_COST_PER_MILE
+  }
```

Comment:
Can you define a `defaultCosts` variable instead of writing out each variable individually?